sandbox/cgroup/tracking.go

```diff
@@ -0,0 +1,282 @@
+package cgroup
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/godbus/dbus"
+
+	"github.com/snapcore/snapd/dbusutil"
+	"github.com/snapcore/snapd/features"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/randutil"
+)
+
+var osGetuid = os.Getuid
+var osGetpid = os.Getpid
+var cgroupProcessPathInTrackingCgroup = ProcessPathInTrackingCgroup
+
+var ErrCannotTrackProcess = errors.New("cannot track application process")
+
+// CreateTransientScopeForTracking puts the current process in a transient scope.
+//
+// To quote systemd documentation about scope units:
+//
+// >> Scopes units manage a set of system processes. Unlike service units,
+// >> scope units manage externally created processes, and do not fork off
+// >> processes on its own.
+//
+// Scope names must be unique, a randomly generated UUID is appended to the
+// security tag, further suffixed with the string ".scope".
+func CreateTransientScopeForTracking(securityTag string) error {
+	if !features.RefreshAppAwareness.IsEnabled() {
+		return nil
+	}
+	logger.Debugf("creating transient scope %s", securityTag)
+
+	// Session or system bus might be unavailable. To avoid being fragile
+	// ignore all errors when establishing session bus connection to avoid
+	// breaking user interactions. This is consistent with similar failure
+	// modes below, where other parts of the stack fail.
+	//
+	// Ideally we would check for a distinct error type but this is just an
+	// errors.New() in go-dbus code.
+	isSessionBus, conn, err := sessionOrMaybeSystemBus(osGetuid())
+	if err != nil {
+		return ErrCannotTrackProcess
```

Comment:
I'll leave this as-is for now but I agree that conceptually it would be good to use xerrors more and that this specific case is an interesting case study.